### ISSUE TITLE
`array_expr_02.py` does not work when imported

### ISSUE THREAD

**Issue by rebcabin** [lcompilers/lpython#2066]
I have a copy of `array_expr_02.py`. The copy is named `hdc.py`. It runs fine. 

I now have a new module called `hdc_test.py`. It has one line in it.

```python
    from hdc import f
```

It generates the following error message:

```bash
/Users/brian/CLionProjects/lpython/src/bin/python /Users/brian/CLionProjects/lpython/lasr/LP-pycharm/hdc_test.py 
warning: The module 'hdc' located in /Users/brian/CLionProjects/lpython/lasr/LP-pycharm/hdc.py cannot be loaded
 --> /Users/brian/CLionProjects/lpython/lasr/LP-pycharm/hdc_test.py:1:1
  |
1 | from hdc import f
  | ^^^^^^^^^^^^^^^^^ imported here

semantic error: Symbol is already declared in the same scope
 --> /Users/brian/CLionProjects/lpython/lasr/LP-pycharm/hdc.py:5:1
  |
5 | n = TypeVar("n")
  | ~~~~~~~~~~~~~~~~ original declaration
  |
4 | n: i32
  | ^^^^^^ redeclaration
```

**Comment by rebcabin** [comment]
The import works if i change the first couple of lines of `hdc.py` from

```python
from lpython import i32, f32, TypeVar
from numpy import empty, sqrt, float32

n: i32
n = TypeVar("n")
```

to

```python
from lpython import i32, f32, TypeVar
from numpy import empty, sqrt, float32

n : i32 = TypeVar("n")
```

**Comment by Smit-create** [comment]
Yes, I also came across the same error while working on one of the PRs. Please remove the following line completely
https://github.com/lcompilers/lpython/blob/e642072180aaa4cb94f27003329430c7ea6d4cce/integration_tests/array_expr_02.py#L4

And that should fix the issue. It's a wrongly written test. But can you please try the above solution by removing line 4 completely and just keeping:
```
n = TypeVar("n")
```

**Comment by rebcabin** [comment]
it works with `n : i32 = TypeVar("n")`. It also works with `n =
TypeVar("n")`. I don't like the second one because it goes against the
pattern of requiring type declarations for all variables. It seems it is a
unique case of type inference, and I thought we do not have any kind of
type inference. There is an open issue about type inference that would
allow `n = TypeVar("n")`, so something is askew :)

**Comment by certik** [comment]
This is an unfortunate inconsistency, since we do not actually support `n` like a variable, it's just for typing purposes to workaround the limitations of CPython. I woudn't even say `n` is i32, I don't think it is. It's not an integer. So `n : i32 = TypeVar("n")` should really say type mismatch. For now, we are just "looking the other way" and pretend we don't see this.

@rebcabin how would you fix this?
